core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -567,6 +568,52 @@ default EventExecutor defaultSubscriberExecutor() {
         return FuseableStreamMessage.error(this, function);
     }
 
+    /**
+     * Peeks values emitted by this {@link StreamMessage} and applies the specified {@link Consumer}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+     * StreamMessage<Integer> ifEvenExistsThenThrow = source.peek(x -> {
+     *      if (x % 2 == 0) {
+     *          throw new IllegalArgumentException();
+     *      }
+     * });
+     * }</pre>
+     */
+    default StreamMessage<T> peek(Consumer<? super T> action) {
+        requireNonNull(action, "action");
+        final Function<T, T> function = obj -> {
+            action.accept(obj);
+            return obj;
+        };
+        return map(function);
+    }
+
+    /**
+     * Peeks values emitted by this {@link StreamMessage} and applies the specified {@link Consumer}.
```

Comment:
Can we also add the following line? I think we should at least document the behavior

```
Only values which are an instance of the specified {@code type} are peeked.
```

Reply:
(I updated it! 😄 )